src/NewsService.Business/GetNewsCommand.cs

```diff
@@ -15,29 +31,147 @@ public class GetNewsCommand : IGetNewsCommand
+      {
+        _logger.LogError("Exception on get department request. {errorsMessage}", exc.Message);
+      }
+
+      errors.Add($"Can not get department info for DepartmentId '{departmentId}'. Please try again later.");
+
+      return null;
+    }
+
+    private ImageData GetAuthorAvatarImage(Guid? imageId, List<string> errors)
+    {
+      try
+      {
+        IOperationResult<IGetImagesResponse> response = _rcGetImages.GetResponse<IOperationResult<IGetImagesResponse>>(
+          IGetImagesRequest.CreateObj(new List<Guid> { imageId.Value }, ImageSource.News)).Result.Message;
+
+        if (response.IsSuccess)
+        {
+          return response.Body.ImagesData.FirstOrDefault();
+        }
+
+        _logger.LogWarning(
+          "Can not get image. Reason:{errors}", string.Join('\n', response.Errors));
+      }
+      catch (Exception exc)
+      {
+        _logger.LogError("Exception on get image request. {errorsMessage}", exc.Message);
+      }
+
+      errors.Add($"Can not get image info for imageId '{imageId}'. Please try again later.");
+
+      return null;
+    }
 
     public GetNewsCommand(
       INewsRepository repository,
       INewsResponseMapper mapper,
-      IHttpContextAccessor httpContextAccessor)
+      IHttpContextAccessor httpContextAccessor,
+      IRequestClient<IGetDepartmentsRequest> rcGetDepartments,
+      IRequestClient<IGetUsersDataRequest> rcGetUsers,
+      IRequestClient<IGetImagesRequest> rcGetImages,
+      IDepartmentInfoMapper departmentMapper,
+      ILogger<GetNewsCommand> logger)
     {
       _repository = repository;
       _mapper = mapper;
       _httpContextAccessor = httpContextAccessor;
+      _rcGetDepartments = rcGetDepartments;
+      _rcGetUsers = rcGetUsers;
+      _rcGetImages = rcGetImages;
+      _departmentInfoMapper = departmentMapper;
+      _logger = logger;
     }
 
     public OperationResultResponse<NewsResponse> Execute(Guid newsId)
     {
       OperationResultResponse<NewsResponse> response = new();
 
-      response.Body = _mapper.Map(_repository.Get(newsId));
+      DbNews dbNews = _repository.Get(newsId);
+      if (dbNews == null)
+      {
+        _httpContextAccessor.HttpContext.Response.StatusCode = (int)HttpStatusCode.BadRequest;
+
+        response.Errors = new() { "News was not found." };
+        response.Status = OperationResultStatusType.Failed;
+        return response;
+      }
+
+      DepartmentInfo department = dbNews.DepartmentId.HasValue ? GetDepartment(dbNews.DepartmentId, response.Errors) : null;
+
+      UserData author = GetAuthor(dbNews.AuthorId, response.Errors);
+      Guid? imageId = author?.ImageId;
+      ImageData avatarImage = GetAuthorAvatarImage(imageId, response.Errors);
+
+      response.Body = _mapper.Map(dbNews, department, author, avatarImage);
+      response.Status = response.Errors.Any()
+        ? OperationResultStatusType.PartialSuccess
+        : OperationResultStatusType.FullSuccess;
+
       if (response.Body == null)
```

Comment:
you check if news not found by id in 147 line, here it is not notFound

Reply:
done